k8s/provider_test.go

```diff
@@ -351,3 +352,108 @@ func marshalPB(obj interface{}) ([]byte, error) {
 	copy(d[len(magicBytes):], body)
 	return d, nil
 }
+
+func TestGenerateServicesWithPropperHealthCheck(t *testing.T) {
+	setEnv(t, "testdata/port_definitions_probe_and_service_only.json")
+	defer unsetEnv(t)
+
+	pod := testPod()
+	path := "/status/ping"
+	pod.Spec.Containers[0].LivenessProbe = &corev1.Probe{
+		Handler: &corev1.Handler{
+			HttpGet: &corev1.HTTPGetAction{
+				Path: &path,
+			},
+		},
+	}
+
+	services, err := generateServices("serviceName", pod, nil)
+	require.NoError(t, err)
+
+	expectedServices := []consul.ServiceInstance{
```

Comment:
Move this above `generateServices` call, as technically this is a setup/given. You could also use //given, //when, //then comments to separate these code blocks.